src/lvm.c

```diff
@@ -1222,8 +1222,11 @@ void luaV_execute (lua_State *L, CallInfo *ci) {
     vmfetch();
 #ifdef USE_YK
     YkLocation *ykloc = NULL;
-    if (GET_OPCODE(i) == OP_FORLOOP || GET_OPCODE(i) == OP_TFORLOOP)
+    if ((GET_OPCODE(i) == OP_FORLOOP) ||
```

Comment:
Right now this is fine *but* it's probably not what we want in the medium term because it introduces several sort-of-constant comparisons that will have an overhead on the interpreter (*probably* the JIT removes all of this).

We've got at least two possible ways here:

1. Rely on `yklocs[pc]` being NULL/non-NULL for the right opcodes. Downside: we have to do a load for every instruction (which might obliterate any advantage we get from removing the comparisons).
2. Is there a spare bit in an opcode we can reuse for "this particular opcode should have a non-null YkLocation"?

Reply:
Honestly, I just did the smallest change possible to get while loops traced.

We used to do `1.`, but that seems to have been changed at some point: https://github.com/ykjit/yklua/blob/51b872ec9b04a2edc27db9519aa4ab38c40d0644/src/lvm.c#L1181

`2.` I couldn't say without studying lua more.